notebooks/refacto.ipynb

```diff
@@ -0,0 +1,1822 @@
+            "text/plain": [
+              "<Figure size 720x720 with 2 Axes>"
+            ]
+          },
+          "metadata": {
+            "tags": [],
+            "needs_background": "light"
+          }
+        },
+        {
+          "output_type": "stream",
+          "text": [
+            "time: 5.69 s (started: 2021-06-02 14:52:31 +00:00)\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "59EsZKHmbGgF"
+      },
+      "source": [
+        "# Decision Tree Regressor\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "CRn8xBCJ9wqA",
+        "outputId": "d27cadd6-5fe0-40e2-a804-91baaf720c97"
+      },
+      "source": [
+        "class DecisionTree(Model):\n",
+        "    \"\"\"\n",
+        "    Class to train and tune decison tree regressor model.\n",
+        "    \"\"\"\n",
+        "\n",
+        "    def __init__(self, params):\n",
+        "        \"\"\"\n",
+        "        Init class with model hyperparameters.\n",
+        "        \"\"\"\n",
+        "        Model.__init__(self, params)\n",
+        "        self.model_name = \"decision_trees\"\n",
+        "\n",
+        "        self.best_max_depth = None\n",
+        "        self.best_min_samples_split = None\n",
+        "        self.min_samples_leaf = None\n",
+        "        self.min_weight_fraction_leaf = None\n",
+        "        self.max_features = None\n",
+        "        self.max_leaf_nodes = None\n",
+        "        self.best_grid = None\n",
+        "        self.best_splitter = None\n",
+        "\n",
+        "        self.run()\n",
+        "\n",
+        "\n",
+        "    def preprocessing(self):\n",
+        "        \"\"\"\n",
+        "        Preprocessing dataset, includes encoding categorical features using a frequency encoder,\n",
+        "        dropping useless features and selecting the target.\n",
+        "        \"\"\"\n",
+        "        self.logger.info('Starting Decision tree preprocessing...')\n",
+        "        encoding = self.dataset.groupby('code_departement').size()\n",
+        "        encoding = encoding/len(self.dataset)\n",
+        "        self.dataset['code_departement'] = self.dataset.code_departement.map(encoding)\n",
+        "\n",
+        "        super().preprocessing()\n",
+        "\n",
+        "    \n",
+        "    def parameter_tuning(self):\n",
+        "        \"\"\"\n",
+        "        Hyperparameter tuning using grid search.\n",
+        "        \"\"\"\n",
+        "        self.logger.info('Starting hyperparameter tuning...')\n",
+        "        # Grid parameters \n",
+        "        params = {\n",
```

Comment:
c'est un peu trop générique, "params", alors que tu as déjà un self.params, je trouve. ça serait pas mieux de l'appeler grid_search_params ou un truc du genre ?

Reply:
Yes, it would be less confusing! Done!